code/modules/admin/admin_verbs.dm

```diff
@@ -355,6 +356,38 @@ GLOBAL_LIST_INIT(admin_verbs_admin, list(
 	if (holder)
 		holder.Secrets()
 
+/client/proc/fix_air()
+	set name = "Fix air (lags)"
+	set category = "Admin"
+	ASSERT(holder)
+
+	// Strip zone of all the air it got from turfs,
+	// reset air in each individual tile, zone itself,
+	// and put now updated air back into the zone's air
+
+	// Iterating through 'world' is bad, but zones are slow to update and their
+	// 'contents' list often does not contain some tiles that must get a reset too
+	for(var/turf/turf in world)
+		if(turf.zone)
+			turf.zone.remove(turf) // Handles visual updates and a part of fire removal
+		if(turf.air)
+			turf.reset_air()
+
+	for(var/zone/zone in SSair.zones)
+		// Often zone's air ends up with more gas than was put in,
+		// probably due to rounding errors or bad volume/temperature math
+		// If left alone, extra gas will leak to the individual turfs
+		// Sometimes that gas is plasma, and/or there is a lot of it
+		qdel(zone.air)
+		zone.air = new
```

Comment:
why not instantiate a new object straight away ? if we bother qdeling the list we just waste CPU on the spot to have something deleted immediately. or were you able to confirm that this always hard-deletes and references have to be looked for in the entire world?

Reply:
(Addendum , DM does have passive garbage collecting for objects with no references)

Reply:
You mean have I tested with and without `qdel()` here to make sure whether it is absolutely necessary or not?
Nah, I didn't.
Just thought since that datum got it's own `Destroy()`, it probably should call it.

Reply:
Fair point , i failed to realise that the Del functionality clients have doesnt also apply to objects, clients just call destroy before del inside of the del proc.. but atoms dont.